qemu/lib/iozone.py

```diff
@@ -0,0 +1,129 @@
+
+    def before_run_iozone(self):
+        """
+        Prepare for iozone test
+        """
+        raise NotImplementedError()
+
+    def run_iozone_in_guest(self, mountpoint):
+        """
+        Run iozone test in guest
+
+        :param mountpoint: mountpoint/disk letter assigned to disk under test
+        :return: iozone run status and result.
+        """
+        iozone_test_cmd_default = "./iozone -azR -r 64k -n 125M -g 512M -i 0 -i 1 %s"
+        iozone_test_cmd = self.params.get("iozone_test_cmd", iozone_test_cmd_default)
+        if self.params.get("os_type") == "windows":
+            test_cmd = iozone_test_cmd % mountpoint
+            test_cmd = utils_misc.set_winutils_letter(self.session, test_cmd)
+        elif self.params.get("os_type") == "linux":
+            test_cmd = iozone_test_cmd % mountpoint
+            self.session.cmd_output_safe('cd %s' % self.iozone_work_path)
+
+        status, output = self.session.cmd_status_output(test_cmd,
+                                                        timeout=self.timeout)
+        self.results_path = os.path.join(self.test.resultsdir, 'raw_output_%s'
+                                         % self.test.iteration)
+        logging.info("Write results to %s" % self.results_path)
+        #omit write results error
+        try:
+            with open(self.results_path, 'w') as file:
+                file.write(output)
+        except:
+            pass
+
+        return status, output
+
+    def after_run_iozone(self):
+        """
+        Clean up iozone env/post result after iozone test
+        """
+        raise NotImplementedError()
+
+
+class Iozone_Win(IozoneTest):
+    """
+    Post result after iozone test.
+    """
+    def __init__(self, vm, params, test):
+        super(Iozone_Win, self).__init__(vm, params, test)
+
+    def after_run_iozone(self):
+        """
+        Post result after iozone test, generate a series graphs
+        """
+        # use this function, params "-i" for iozone is not suggested.
+        # suggest usage:iozone.exe -azR -r 64k -n 125M -g 512M -M -b
+        # I:\iozone.xls -f I:\testfile".
+        # Otherwise, your result can't be analysed correctly.
+        if self.params.get("post_results", "no") == "yes":
+            logging.info("Generate graph of test result")
+            analysisdir = os.path.join(self.test.resultsdir, 'analysis_%s'
+                                       % self.test.iteration)
+            iozone_analyzer = postprocess_iozone.IOzoneAnalyzer(list_files=[self.results_path],
+                                                                output_dir=analysisdir)
+            iozone_analyzer.analyze()
+            iozone_plotter = postprocess_iozone.IOzonePlotter(results_file=self.results_path,
+                                                              output_dir=analysisdir)
+            iozone_plotter.plot_all()
+
+
+class Iozone_Linux(IozoneTest):
```

Comment:
@aliang123 we can configure and reuse VMStress() available in virttest/libvirt/__init__.py ? Is there any specific things to be taken care ?

Reply:
@balamuruhans, I will check it, and give a reply later.

Reply:
@balamuruhans, i can't find virttest/libvirt/init.py, do you mean virttest/utils_test/__init__.py?

Reply:
I have reworked on the VMStress API, you can refer to https://github.com/avocado-framework/avocado-vt/pull/1598.
I would request your review and suggestions on it. Thank you.

Reply:
@balamuruhans, VMStress() in virttest/utils_test/__init__.py can only applied on linux guest. This patch will combine linux guest and windows guest together, so i called VMStress() in class Iozone_Linux().__init__(). Or you suggest Multiple inheritance for class Iozone_Linux()?

Reply:
ok, got it, i'll check.